src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -375,19 +383,30 @@ int xma_xclbin_info_get(const char *buffer, XmaXclbinInfo *info)
     return XMA_SUCCESS;
 }
 
-int xma_xclbin_map2ddr(uint64_t bit_map, int32_t* ddr_bank)
+int xma_xclbin_map2ddr(uint64_t bit_map, int32_t* ddr_bank, bool has_mem_grps)
 {
     //64 bits based on MAX_DDR_MAP = 64
     int ddr_bank_idx = 0;
-    while (bit_map != 0)
-    {
-        if (bit_map & 1)
-        {
-            *ddr_bank = ddr_bank_idx;
-            return XMA_SUCCESS;
+    if (!has_mem_grps) {
```

Comment:
Often XRT lacks some high-level comments...
Not clear if this is correct since there is no explanation to compare with.
It looks like your are parsing a bitset here to find the number of trailing 0...
You need some C++20 here https://en.cppreference.com/w/cpp/numeric/countr_zero

Reply:
Correct. Leading and trailing zeros are checked in two mem_topologies. C++20 is not yet supported by XRT. Good suggestion.